library/src/main/java/com/google/android/exoplayer2/text/eia608/Eia608Decoder.java

```diff
@@ -414,32 +501,197 @@ private void handleMiscCode(byte cc2) {
       case CTRL_BACKSPACE:
-        if (captionStringBuilder.length() > 0) {
-          captionStringBuilder.setLength(captionStringBuilder.length() - 1);
-        }
+        backspace();
         return;
     }
   }
 
+  private void handlePreambleCode(byte cc1, byte cc2) {
+    // For PAC layout see: https://en.wikipedia.org/wiki/EIA-608#Control_commands
+    applySpan(); // Apply any spans.
+
+    // Parse the "next row down" flag.
+    boolean nextRowDown = (cc2 & 0x20) != 0;
+    if (nextRowDown) {
+      // TODO: We should create a new cue instead, this may cause issues when
+      // the new line receives it's own PAC which we ignore currently.
+      // As a result of that the new line will be positioned directly below the
+      // previous line.
+      maybeAppendNewline();
+    }
+
+    // Go through the bits, starting with the last bit - the underline flag:
+    boolean underline = (cc2 & 0x1) != 0;
+    if (underline) {
+      captionStyles.put(getSpanStartIndex(), new UnderlineSpan());
+    }
+
+    // Next, parse the attribute bits:
+    int attribute = cc2 >> 1 & 0xF;
+    if (attribute >= 0x0 && attribute < 0x7) {
+      // Attribute is a foreground color
+      captionStyles.put(getSpanStartIndex(), new ForegroundColorSpan(COLOR_MAP[attribute]));
+    } else if (attribute == 0x7) {
+      // Attribute is "italics"
+      captionStyles.put(getSpanStartIndex(), new StyleSpan(STYLE_ITALIC));
+    } else if (attribute >= 0x8 && attribute <= 0xF) {
+      // Attribute is an indent
+      if (cueIndent == DEFAULT_INDENT) {
+        // Only update the indent, if it's the default indent.
+        // This is not conform the spec, but otherwise indentations may be off
+        // because we don't create a new cue when we see the nextRowDown flag.
+        cueIndent = INDENT_MAP[attribute & 0x7];
+      }
+    }
+
+    // Parse the row bits
+    int row = cc1 & 0x7;
+    if (row >= 0x4) {
+      // Extended Preamble Code
+      row = row & 0x3;
+      switch (row) {
+        case 0x0:
+          // Row 14 or 15
+          cueLine = CUE_LINE_MAP[13];
+          break;
+        case 0x1:
+          // Row 5 or 6
+          cueLine = CUE_LINE_MAP[4];
+          break;
+        case 0x2:
+          // Row 7 or 8
+          cueLine = CUE_LINE_MAP[7];
+          break;
+        case 0x3:
+          // Row 9 or 10
+          cueLine = CUE_LINE_MAP[8];
+          break;
+      }
+    } else {
+      // Regular Preamble Code
+      switch (row) {
+        case 0x0:
+          // Row 11 (Default)
+          cueLine = CUE_LINE_MAP[10];
+          break;
+        case 0x1:
+          // Row 1 (Top)
+          cueLine = CUE_LINE_MAP[0];
+          break;
+        case 0x2:
+          // Row 4 (Top)
+          cueLine = CUE_LINE_MAP[3];
+          break;
+        case 0x3:
+          // Row 12 or 13 (Bottom)
+          cueLine = CUE_LINE_MAP[11];
+          break;
+      }
+    }
+  }
+
+  private void handleMidrowCode(byte cc1, byte cc2) {
```

Comment:
This should take care of the order of priorities: https://github.com/google/ExoPlayer/pull/1773/commits/7cd94819c7a4a0222022cbfd49a1d77f30aa5812#diff-7fa75103163e987aeb1657e153a4604fR596

Do you by any chance have a stream that uses midrow codes to change styles? Apple's HLS sample doesn't seem to do it.